design/kata-api-design.md

```diff
@@ -0,0 +1,143 @@
+# Kata API Design
+To fulfill the [kata design requirements](kata-design-requirements.md), and based on the disscusion on [Virtcontainers API extentions](https://docs.google.com/presentation/d/1dbGrD1h9cpuqAPooiEgtiwWDGCYhVPdatq7owsKHDEQ), the kata runtime library features following APIs:
+-  Sandbox based top API
+-  Storage and network hotplug API
+-  Plugin frameworks for external proprietary Kata runtime extensions
+-  Built-in shim and proxy types and capabilities
+
+## Sandbox Based API
+### Sandbox Management API
+-  CreateSandbox(ID, SandboxConfig)
```

Comment:
Is there a reason none of the listed APIs show parameter types and return values? It is difficult to know how the API might behave without these imho.

Reply:
Sorry I missed this one. Yes, there is a reason -- I wanted to focus on the API set and workflow first. API parameter types/return values can be debatable for a long time and I'm deferring them to implementation PRs ;)